lib/Sema/TypeCheckStorage.cpp

```diff
@@ -1572,6 +1570,19 @@ synthesizeCoroutineAccessorBody(AccessorDecl *accessor, ASTContext &ctx) {
                    ? TargetImpl::Ordinary
                    : TargetImpl::Implementation);
 
+  // If this is a variable with an attached property wrapper, then
+  // the accessors need to yield the wrappedValue or projectedValue.
+  if (auto var = dyn_cast<VarDecl>(storage)) {
+    if (var->hasAttachedPropertyWrapper()) {
+      target = TargetImpl::Wrapper;
+    }
+
+    if (var->getOriginalWrappedProperty(
+            PropertyWrapperSynthesizedPropertyKind::StorageWrapper)) {
+      target = TargetImpl::WrapperStorage;
+    }
+  }
+
```

Comment:
@theblixguy I don't understand why this is required.

Given a property like:
~~~
struct Foo {
  @Wrap var bar: Int
}
~~~

Before this PR, the synthesized modify in SIL looked like:
~~~
Foo.bar.modify:
  var tmp = call Foo.bar.getter(self)
  yield &tmp
  call Foo.bar.setter(tmp, &self)
~~~

where the Foo.text.getter calls Wrap.wrappedValue.getter, and Foo.text.setter calls Wrap.wrappvedValue.setter.

After this PR, it looks like:
~~~
Foo.bar.modify:
  var tmp = call Wrap.wrappedValue.getter(self._text)
  yield &tmp
  call Wrap.wrappedValue.setter(self._text)
~~~

These are equivalent, but I thought it was cleaner that Foo.bar.modify calls into Foo.bar.getter/setter instead of being doing something special for property wrappers. Why is it being done this way?